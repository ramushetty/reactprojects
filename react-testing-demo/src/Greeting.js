import React from 'react'

export const Greeting = ({name = 'Guest'}) => {
  return (
    <h1>
        Hello, {name}!
    </h1>
  )
}
