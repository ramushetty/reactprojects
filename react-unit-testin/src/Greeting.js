// Greeting.js
import React from 'react';

function Greeting({ name = 'Guest' }) {
  return <h1>Hello, {name}!</h1>;
}

export default Greeting;
