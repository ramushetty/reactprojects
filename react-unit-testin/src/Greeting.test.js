// Greeting.test.js
import React from 'react';
import { render, screen } from '@testing-library/react';
import Greeting from './Greeting';

test('renders greeting message with name', () => {
  render(<Greeting name="John" />);
  expect(screen.getByText('Hello, John!')).toBeInTheDocument();
});

test('renders default greeting when no name is provided', () => {
  render(<Greeting />);
  expect(screen.getByText('Hello, Guest!')).toBeInTheDocument();
});


it('renders correctly with a name', () => {
    const { asFragment } = render(<Greeting name="John" />);
    expect(asFragment()).toMatchSnapshot();
  });
  
it('renders correctly without a name', () => {
const { asFragment } = render(<Greeting />);
expect(asFragment()).toMatchSnapshot();
});



