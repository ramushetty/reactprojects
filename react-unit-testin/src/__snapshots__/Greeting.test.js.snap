// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly with a name 1`] = `
<DocumentFragment>
  <h1>
    Hello, John!
  </h1>
</DocumentFragment>
`;

exports[`renders correctly without a name 1`] = `
<DocumentFragment>
  <h1>
    Hello, Guest!
  </h1>
</DocumentFragment>
`;
